src/groundlight/images.py

```diff
@@ -1,10 +1,35 @@
 import imghdr
-from io import BufferedReader, BytesIO
+from io import BufferedReader, BytesIO, IOBase
 from typing import Union
 
 from groundlight.optional_imports import Image, np
 
 
+class ByteStreamWrapper(IOBase):
+    """This class acts as a thin wrapper around bytes in order to
+    maintain files in an open state. This is useful, in particular,
+    when we want to retry accessing the file without having to re-open it.
+    """
+
+    def __init__(self, data: Union[BufferedReader, BytesIO, bytes]) -> None:
+        super().__init__()
+        self._data = data
+
+    def read(self) -> bytes:
+        if isinstance(self._data, (BufferedReader, BytesIO)):
```

Comment:
I don't think this works.  Because if you call `.read()` twice in a row on a BytesIO object, the second time, you get back `b''` - empty.  I think you should do this conversion (calling `read()`)in `__init__` to ensure you only call read() once.  And then read & getvalue can just return data.